SingleCellPatch/extract_patches.py

```diff
@@ -322,51 +209,103 @@ def process_site_extract_patches(site_path,
-        cell_segmentation = segmentation_stack[t_point, :, :]
-        
+        raw_image = image_stack[t_point]
+        cell_segmentation = segmentation_stack[t_point]
+        cell_segmentation = check_segmentation_dim(cell_segmentation)
         positions, positions_labels = cell_pixel_assignments[t_point]
         mg_cells, non_mg_cells, other_cells = cell_positions[t_point]
-
+        all_cells = mg_cells + non_mg_cells + other_cells
         # Define fillings for the masked pixels in this slice
         background_pool = raw_image[np.where(cell_segmentation[:, :, 0] > 0.9)]
         background_pool = np.median(background_pool, 0)
         background_filling = np.ones((window_size, window_size, 1)) * background_pool.reshape((1, 1, -1))
-
+        cells_to_keep = []
         # Save all cells in this step, filtering will be performed during analysis
-        for cell_id, cell_position in mg_cells + non_mg_cells + other_cells:
+        for cell_id, cell_position in all_cells:
             cell_name = os.path.join(site_supp_files_folder, '%d_%d.h5' % (t_point, cell_id))
             if cell_name in site_data:
                 continue
             # Define window based on cell center and extract mask
             window = [(cell_position[0]-window_size//2, cell_position[0]+window_size//2),
                       (cell_position[1]-window_size//2, cell_position[1]+window_size//2)]
-            window_segmentation = select_window(cell_segmentation, window, padding=-1)
+            # window_segmentation = select_window(cell_segmentation, window, padding=-1)
+            window_segmentation = select_window(cell_segmentation, window, padding=None)
+            if window_segmentation is None:
+                continue
+            # only keep the cells that has patches
+            cells_to_keep.append(cell_id)
             remove_mask, tm, tm2 = generate_mask(positions, 
                                                  positions_labels, 
                                                  cell_id, 
                                                  window, 
                                                  window_segmentation)
             # Select submatrix from the whole slice
-            output_mat = select_window(raw_image, window, padding=0)
+            # output_mat = select_window(raw_image, window, padding=0)
+            output_mat = select_window(raw_image, window, padding=None)
             masked_output_mat = output_mat * (1 - remove_mask) + background_filling * remove_mask
             # To prevent backward compatibility issue cast to int64 and float 64 respectively
             # TODO: solve compatibility issue here
             output_mat = np.concatenate([output_mat, tm, tm2], 2).astype('int64')
             masked_output_mat = np.concatenate([masked_output_mat, tm, tm2], 2).astype('float64')
             site_data[cell_name] = {"mat": output_mat, "masked_mat": masked_output_mat}
+            if save_fig:
```

Comment:
I have added this function on my end, the flag will be exposed to pipeline functions as well.